### DISCUSSION TITLE
post_233621

### DISCUSSION THREAD

**Discussion by cincuranet** [cincuranet/blog-comments#1251] [Comments]
imported 

https://www.tabsoverspaces.com/id/233621

**Comment by cincuranet** [comment]
Imported comment written by _Kevin Dillard_ on _2017-10-12T13:40:57_:

<p>Hi. Could you please help me a bit - in earlier driver versions there was a method - AddEvents.</p><p>Something like this - revent.AddEvents(new string[] { "new_g" });</p><p>Could you please tell how can i replace it?</p><p>Thanks.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-10-12T13:42:13_:

<p>It's right there, in the example, `QueueEvents`.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Kevin Dillard_ on _2017-10-13T20:00:44_:

<p>Thank you for your reply.</p><p>Am I understanding this correctly - this is how it was:</p><p>revent.AddEvents(new string[] { "new_g" });<br>revent.QueueEvents();</p><p>And now it's become:</p><p>revent.QueueEvents(new string[] { "new_g" });</p><p>Thanks.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-10-13T20:28:22_:

<p>Correct.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Kevin Dillard_ on _2017-10-16T13:03:08_:

<p>Great, thanks!</p>

**Comment by cincuranet** [comment]
Imported comment written by _Onesimo_ on _2019-03-29T16:46:37_:

<p>The sample File FbRemoteEventTests.cs  is no longer avalaible in GitHub.   where can I find it?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2019-03-29T16:55:11_:

<p>It sure is: <a href="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/Provider/src/FirebirdSql.Data.FirebirdClient.Tests/FbRemoteEventTests.cs" rel="nofollow noopener" title="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/Provider/src/FirebirdSql.Data.FirebirdClient.Tests/FbRemoteEventTests.cs">https://github.com/cincuran...</a> .</p>

**Comment by cincuranet** [comment]
Imported comment written by _Szilard M_ on _2019-04-02T23:33:03_:

<p>Hello,<br>Cannot make it work with event posted from trigger. It worked with the previous version just fine.<br>If I post the event from code as you, it works.<br>Usage:</p><p>private FbRemoteEvent remoteEventComanda;<br>remoteEventComanda = new FbRemoteEvent(GetConnectionString());<br>                    remoteEventComanda.RemoteEventCounts += remoteEventComanda_RemoteEventCounts;<br>                    remoteEventComanda.QueueEvents(new[] { "new_comand" });</p><p>void remoteEventComanda_RemoteEventCounts(object sender, FbRemoteEventCountsEventArgs e)<br>        {<br>   //never fires from trigger<br>    Console.WriteLine("Event {0} has {1} counts.", e.Name, e.Counts);<br>    InvokeComandListChanged(e);<br>        }</p><p>Trigger:<br>CREATE OR ALTER TRIGGER COMAND_ANDR_AI0 FOR COMAND_ANDR</p><p>ACTIVE AFTER INSERT POSITION 0<br>AS<br>begin<br>  /* Trigger text */<br>   post_event 'new_comand';<br>end</p><p>Thanks.</p><p>Edit.: My fault! Fixed! Sorry to bother you.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Raheel Ahmed Khan Lodhi_ on _2020-04-02T15:43:05_:

<p>Hello..</p><p>I have a form application in C# Visual Studio 2019 from where I want to listen FB EVENT always whenever there is some new records in a specific able. Can you please help ?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-04-02T17:57:38_:

<p>You need to create triggers to send the events (using `POST_EVENT`) and then register for these events using `FbRemoveEvent` class.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Raheel Ahmed Khan Lodhi_ on _2020-04-02T18:07:56_:

<p>CREATE OR ALTER TRIGGER SHOOTSALE FOR FAMILIES<br>ACTIVE AFTER INSERT POSITION 0<br>AS<br>begin<br>post_event 'SHOOTSALE';<br>end</p><p>this is what i have created in Firebird. But now the issue is that how to listen it in C#. Can you please help me with some source code support ?</p><p>In C#, I have like this:<br>string[] daftarEvent = { "SHOOTSALE" };<br>FbRemoteEvent remoteEvent = new FbRemoteEvent(strCon);</p><p>-- This code line is a problem for me, i do not have any clue how to call the function--<br>remoteEvent.RemoteEventCounts += ?????</p><p>Thank You</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-04-02T18:09:05_:

<p>There's an example in docs (<a href="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/Provider/docs/events.md)" rel="nofollow noopener" title="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/Provider/docs/events.md)">https://github.com/cincuran...</a>.</p>

**Comment by spaloma** [comment]
Hi! I didn't know how to use remoteevent but your article help me so much. Thank you.

**Comment by cincuranet** [comment]
Glad it helped. BTW there's also [docs page for events](https://github.com/FirebirdSQL/NETProvider/blob/master/docs/events.md).

**Comment by stssterg** [comment]
Good day, the links don't work.

Also, no matter how much I've tried I cannot receive any events while running a firebird database and a wpf application on the same PC.
Is there a change in code? I suppose there is cause it won't let me use the code without adding the line @event.Open(). Can you offer any insight? Any help would be much appreciated.

**Reply by stssterg** [reply]
I tried the code on ConsoleApps both Net Framework and Core and it works! But the exact same code does not work in WPF. Any suggestions?

**Reply by cincuranet** [reply]
Probably you're blocking the `SynchronizationContext` and hence the events are not processed.

**Reply by stssterg** [reply]
could you explain a bit more? I have a MVVM WPF application and I run the code in the MainWindowViewModel. Any suggestions on how to make it work?

**Reply by stssterg** [reply]
For anyone who reaches this point searching for a solution: Turns out you need to listen for events on a separate thread. Instead of executing the code above directly in your ViewModel, write an async void function, run the code in there except for the "events.QueueEvents(eventNames" line. Put that inside an "await Task.Run(()=> events.QueueEvents(eventNames));" like that and the call it from your ViewModel constructor. It'll then run smoothly.

Thank you cincuranet for your work and for your quick response even after all of these years. 

Take care

**Reply by stssterg** [reply]
Is there a way to listen for all events?

**Reply by cincuranet** [reply]
No.
